pkg/modulewriter/tfwriter.go

```diff
@@ -193,21 +193,27 @@ type provider struct {
 	config  config.Dict
 }
 
-func getProviders(bp config.Blueprint) []provider {
+func getProviders(bp config.Blueprint, g config.Group) []provider {
 	gglConf := config.Dict{}
-	for s, v := range map[string]string{
-		"project": "project_id",
-		"region":  "region",
-		"zone":    "zone"} {
-		if bp.Vars.Has(v) {
-			gglConf = gglConf.With(s, config.GlobalRef(v).AsValue())
+	if len(g.TerraformProviders) == 0 {
+		for s, v := range map[string]string{
+			"project": "project_id",
+			"region":  "region",
+			"zone":    "zone"} {
+			if bp.Vars.Has(v) {
+				gglConf = gglConf.With(s, config.GlobalRef(v).AsValue())
+			}
+		}
+		return []provider{
+			{"google", "hashicorp/google", ">= 4.84.0, < 5.30.0", gglConf},
```

Comment:
This list of "default providers" need to be moved to `config package` and used as a default value for `Blueprint.TerraformProviders`.

Reply:
Moved.